packages/yarnpkg-pnp/sources/loader/makeApi.ts

```diff
@@ -497,22 +496,16 @@ export function makeApi(runtimeState: RuntimeState, opts: MakeApiOptions): PnpAp
     if (!relativeLocation.endsWith(`/`))
       relativeLocation = `${relativeLocation}/` as PortablePath;
 
-    let from = 0;
+    do {
+      const entry = packageLocatorsByLocations.get(relativeLocation);
 
-    // If someone wants to use a binary search to go from O(n) to O(log n), be my guest
```

Comment:
I tested a binary search but wasn't much of a difference

Reply:
Your binary search is not O(log n) ;-) ?

Reply:
I did this back in December but I think I used https://yarnpkg.com/package/binary-search

Reply:
Tested again with https://github.com/mikolalysenko/binary-search-bounds

```diff
# yarn bin
- 6609 calls, 107.37779996916652 ms
+ 6609 calls, 122.76100007817149 ms
```